### DISCUSSION TITLE
Setting Timeout Option in rustsat-cadical

### DISCUSSION THREAD

**Discussion by gn64** [chrjabs/rustsat#204] [Q&A]
Thank you for sharing this wonderful code. 
I am currently trying to use Cadical from rustsat and would like to ask about how to set options.
I understand that it works in the form of `solver.set_option("seed", 42)`, but how should I set a timeout?
According to `cadical --help`, it should be set like `-t <sec> set wall clock time limit`, but using `solver.set_option("t", 10)` did not work.

**Comment by chrjabs** [comment]
Hi,
I had a quick look at CaDiCaL's source code, and it looks like the time limit is not enforced within CaDiCaL itself, but from the main function, which then terminates CaDiCaL via a terminator (see [here](https://github.com/arminbiere/cadical/blob/31f9aae44947c673dc6190c9beb399acb9ffec6a/src/cadical.cpp#L644)). So the `-t` option applies only to the `main` function of the CaDiCaL _executable_, not to the solving library itself.
To do something similar via RustSAT, you could get an [`Interrupter`](https://docs.rs/rustsat-cadical/latest/rustsat_cadical/struct.CaDiCaL.html#method.interrupter) for the solver and then use some sort of timer for terminating CaDiCaL via the `Interrupter`.
I haven't actually tried to do this myself, so I can't really give you much more than this sketch of an idea.
